### ISSUE TITLE
mojo_run's --disable-cache (and multiple shells on the same machine)

### ISSUE THREAD

**Issue by afandria** [domokit/mojo#477]
I cannot disable caching on `mojo_run` with `--disable-cache`.

My use case is to run 2 mojo applications on the same machine. These applications use their own origin, and thus, exist outside of mojo:FOO. Normally, these will be run on different machines, but for development, it is better to have them on the same one.

Main issues:
* Duplicate ports. It looks like when applications are outside of `mojo:FOO`, they start up a mapping server at a fixed port: 31841 if you use `map-origin`. I decided not to use that to avoid that problem.
* Duplicate ports pt 2. Since I can't use `map-origin`, I had to write my own mojoconfig file to specify the mapping. This will also potentially have duplicate ports, so I used a `--config-alias` to allow me to specify the port in my Makefile commands.
* Cache. It looks like apps outside of `mojo:FOO` are cached by default. Supposedly this can be disabled with `--disable-cache`, but it didn't work.

Unfortunately, my mojo file was still cached anyway. I had to delete `~/.mojo_url_response_disk_cache/db/LOCK` before I could run the second shell.

**Comment by qsr** [comment]
Well, the cache is disabled, in the sense that you do not use it and always do a network request. Now, it doesn't mean that the service doesn't exist, and the service creates a leveldb database that you try to open twice in 2 different processes.

Anything that uses any persistence will have an issue with this setup.

To solve this, you can set the HOME environment variable before running. That should solve your issue.

**Comment by afandria** [comment]
Thanks, so it was a leveldb database instead that was creating this lock.
Prefixing my commands with HOME={different directories per command} works.

**Comment by ppiet** [comment]
So if I read this correctly, what remains here is to provide an option to use system-allocated ports for all dev servers (not only for the host-side in an Android setup as in #470) in `mojo_run`. Let's use this bug to track.

**Comment by ppiet** [comment]
@afandria would it work for you to specify multiple origins/servers in your mojoconfig, one per app instance? Along the lines of:

```
'dev_servers': [
    {
      'host': 'https://app1.io/',
      'port': 0,
      'mappings': [(...)]
    },
    {
      'host': 'https://app2.io/',
      'port': 0,
      'mappings': [(...)]
    },
```

**Comment by afandria** [comment]
I've attempted to reuse the same code for each app, so I think it's actually a harder problem if I specify each dev server this way. (I didn't realize I could pick port 0 though, so that simplifies my current workaround.)

From what I noticed, the issue is that if I use `--map-origin`, I start up a mapping server on the fixed port `31841` (and no flags allow this to be system-allocated). I worked around that duplicate port problem with a mojoconfig file.
